Cassette config inheritance

aaronwolen opened the pull request:
As discussed in #151. This changes the defaults for arguments in `use_cassette()`/`insert_cassette()` to `NULL` *if* they control parameters that are also configurable via `VCRConfig`. Before making this change I reformatted the code so there's only one argument per line to make the [diff](https://github.com/ropensci/vcr/pull/153/commits/3a24c0aa3fa50ab84ccd0f739ba70c346ef71084) more readable.

Other changes: 
* Adds test helper `vcr_test_configuration()` to reset the current configuration and restore settings that should be in place for all tests. This could be called within a `teardown()` block at the top of every test file that includes modifications to the configuration. Currently it's only used in the new inheritance tests.
* Adds tests to verify that cassettes inherit settings from `vcr_configuration()`

sckott commented:
All looks good, except:

We should update the parameter docs for `insert_cassette` and `use_cassette` for those that are changed in this PR to say that their default values are set by `vcr_configure` - which only needs to be done in `use_cassette` i guess since `insert_cassette` inherits from `use_cassette`

sckott commented:
And document - make it explicit that any parameters set in `insert_cassette`/`use_cassette` override those set with `vcr_configure`

aaronwolen commented:
Cool, I'll work on updating the docs.

aaronwolen commented:
I took a stab at documenting the new inheritance behavior in 
6d033a7 but I feel like it could be clearer. That was the best my brain could come up with late on a Friday afternoon 🙂. Let me know if you have any suggestions. Otherwise I'll make another pass on Monday.

sckott commented:
looks good to me - can you resolve the conflict, then i can merge

aaronwolen commented:
Thanks for looking it over. I made a few additional non-coding changes that include some minor rewording/reorgs in the docs.

The most significant is cb23594, which  reorganizes the configuration docs a bit to create a separate section for cassette options that can be configured both globally and locally—and added a brief description that attempts to explain how they relate to each other. 

Also just rebased to sync up with master.

sckott commented:
Both look good, thanks.
